jobsdb/jobsdb.go

```diff
@@ -2058,22 +2058,15 @@ func (jd *HandleT) backupTable(backupDSRange dataSetRangeT, isJobStatusTable boo
 			panic(err)
 		}
 
-		var rows []interface{}
-		err = json.Unmarshal(rawJSONRows, &rows)
-		if err != nil {
-			panic(err)
-		}
-		contentSlice := make([][]byte, len(rows))
-		for idx, row := range rows {
-			rowBytes, err := json.Marshal(row)
-			if err != nil {
-				panic(err)
-			}
-			contentSlice[idx] = rowBytes
+		if len(rawJSONRows) < 2 {
+			break
 		}
-		// append new line character at end, before next backupRowsBatch is appended to same file
-		content := append(bytes.Join(contentSlice[:], []byte("\n")), []byte("\n")...)
-		gzWriter.Write(content)
+
+		rawJSONRows = bytes.Replace(rawJSONRows, []byte(", \n "), []byte("\n"), -1) //replacing ", \n " with "\n"
+		rawJSONRows = rawJSONRows[1 : len(rawJSONRows)-1]                           //stripping starting '[' and ending ']'
```

Comment:
Let's add an assert to ensure the first and last characters are `[` and `]`?

Reply:
json_agg returns a JSON Array. https://www.postgresql.org/docs/9.5/functions-aggregate.html.

If we want an assert, we can add.

Reply:
added assert.